wrappers/services.go

```diff
@@ -104,20 +104,46 @@ func stopService(sysd systemd.Systemd, app *snap.AppInfo, inter interacter) erro
 // StartServices starts service units for the applications from the snap which
 // are services. Service units will be started in the order provided by the
 // caller.
-func StartServices(apps []*snap.AppInfo, inter interacter, tm timings.Measurer) (err error) {
+func StartServices(apps []*snap.AppInfo, disabledSvcs []string, enableBeforeStart bool, inter interacter, tm timings.Measurer) (err error) {
 	sysd := systemd.New(dirs.GlobalRootDir, systemd.SystemMode, inter)
-
 	services := make([]string, 0, len(apps))
+
+	defer func() {
+		if err == nil {
+			return
+		}
+
+		for _, srvName := range services {
+			if e := sysd.Disable(srvName); e != nil {
+				inter.Notify(fmt.Sprintf("While trying to disable previously enabled service %q: %v", srvName, e))
+			}
+		}
+	}()
+
 	for _, app := range apps {
 		// they're *supposed* to be all services, but checking doesn't hurt
 		if !app.IsService() {
 			continue
 		}
 
+		if len(app.Sockets) == 0 && app.Timer == nil {
```

Comment:
If I understand correctly we don't enable services that are supposed to be activated via timer or sockets, can we add a comment here to explain that

Reply:
is the if false path tested for this?

Reply:
They are enabled, but below in separate go. Added a comment.
Yes, we have tests for sockets, timers (and services without them) in services_test.go. These are existing tests that I updated for new semantics.